modules/actions/workflows.go

```diff
@@ -5,9 +5,11 @@ package actions
 
 import (
 	"bytes"
+	"fmt"
 	"io"
 	"strings"
 
+	actions_model "code.gitea.io/gitea/models/actions"
```

Comment:
`modules` is the wrong place for this.
Go does not allow import cycles.
Please do not add any imports to `models` from `modules`.

Reply:
@delvh this action module already import the actions model. 

https://github.com/go-gitea/gitea/blob/main/modules/actions/task_state.go#L7

My understanding from reading the codebase is that modules import models, services import models and modules, routers import models, modules, and services. Can you explain how I should be thinking about this or point me to docs on it?

Reply:
> This module is already dependent on the actions model here: https://github.com/go-gitea/gitea/blob/main/modules/actions/task_state.go#L7

:vomiting_face: Okay, that is one of the places we didn't fix yet.
We are still working on removing all the historically grown errors.
For the time being, I think it is better not to introduce any already known technical debt.

As to where to add it, I would have added it as a type method on `actions_model.ActionRun` which is defined in `models/actions/run.go`